utils/async/PendingTaskModel.js

```diff
@@ -19,15 +19,16 @@ import {isUndefined} from 'lodash';
  * @see Promise#linkTo
  * @see Panel.mask
  */
-@HoistModel
 export class PendingTaskModel {
```

Comment:
Should we extend `HoistModel` here?

Reply:
We cannot.  Util is a lower level package, even lower level than  core.  The PendingTaskModel is actually built in to AppBase, so you would run in to a circular reference problems.

In retrospect the name PendingTaskModel was probably a mistake.  Should have just been 'PendingTask' and there would be no confusion.

Note that PendingTaskModel actually still serves as a `model` for the mask()!  It's a point that we don't document, but there was *never* actually a requirement that the model behind any component is actually a HoistModel, and this PR continues with that flexibility and actually extends it, allowing any 'model' to be looked up/placed in context as well.

 It's a useful feature -- we have one or two examples in toolbox of components that use XH as their model.   I would be surprised if applications do this as well, and I think its an important bit of flexibility -- but something we need to discuss.  HoistModel itself is just an (important) convention in the toolkit right now for easily getting an object that has the support of HoistBase. Let's discuss if we are ok with this, and how to best document it.

Reply:
Got it, thanks.  I recall the early discussions around what can qualify as a "model," and I still agree that it's a feature to have that be wide open.  Just as you could have a component that's not a product of the hoistComponent factory, you might have something called a model that doesn't extend `HoistModel`.